core/lib/rom/schema/attribute.rb

```diff
@@ -369,6 +369,32 @@ def respond_to_missing?(name, include_private = false)
         type.respond_to?(name) || super
       end
 
+      # Return AST for the type
+      #
+      # @return [Array]
+      #
+      # @api public
+      def to_ast
+        @__ast__ ||= [:attribute, [name, type.to_ast(meta: false), meta_ast]]
+      end
+
+      # Return AST for the read type
+      #
+      # @return [Array]
+      #
+      # @api public
+      def to_read_ast
+        @__read_ast__ ||= [:attribute, [name, to_read_type.to_ast(meta: false), meta_ast]]
+      end
+
+      # @api private
+      def meta_ast
+        meta_keys = %i(wrapped alias primary_key)
```

Comment:
@solnic fyi I had to add `:primary_key` and `:source` because of https://github.com/rom-rb/rom/blob/8aa0b411e3c4561e850459812b0723b60f692a31/repository/lib/rom/repository/changeset/associated.rb#L28-L38